### PR TITLE
Restoring the return type of the Expression<TDelegate>.Update

### PR DESCRIPTION
**Pull request by VSadov**
The method is a public API and is used to return Expression< TDelegate >

### CODE REVIEW

File: src/System.Globalization/tests/project.lock.json [hunk 1]
```diff
@@ -3,6 +3,17 @@
   "version": -9996,
   "targets": {
     "DNXCore,Version=v5.0": {
+      "System.AppContext/4.0.0-beta-23024": {
```
Comment (akoeplinger) [inline src/System.Globalization/tests/project.lock.json]: The changes to this file aren't related to your change and looks like they were missed when the globalization tests were added, I think we should separate them out into a distinct commit/PR.

Comment (stephentoub) [inline src/System.Globalization/tests/project.lock.json]: Agreed.

Comment (VSadov) [inline src/System.Globalization/tests/project.lock.json]: The changes happen automatically when I build. I was not sure if they are needed. I can exclude them.

**Comment by VSadov** [comment]
CC @weshaggard @Pilchie

**Comment by weshaggard** [comment]
cc @nguerrera 
Is this going to be compatible? What did we change from the desktop to require this to be a newslot?

**Comment by nguerrera** [comment]
Why is dropping override OK? Is anything calling through the base Update method? The base method is not part of the public surface seen through contracts. If nothing is using it, it should be removed. If something is, then it needs to be renamed (can't overload on return type) and overridden. In the latter case, I'd also mark them internal to draw attention to the fact that this is implementation-only.

Are there tests covering this? I realize it may be source compatible with them (stronger return type than before). Perhaps it's worth making the tests require `Expression<TDelegate>` in some way?

**Comment by VSadov** [comment]
@nguerrera - you are right, the change is more involved than just changing the Update. I also need to remove Update from the LambdaExpressin (thus making newslot unnecessary).

I think the whole thing might have been added as a workaround for generic virtual Visit that takes Expression<T>, but it appears that it is still there, so I am not sure. 
Will need to check if this change is ok with .Net native before committing.

**Comment by nguerrera** [comment]
There should also have been a test that failed when the override became newslot. Can you make sure that stress that code path?

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by VSadov** [comment]
@dotnet-bot retest this please

**Comment by VSadov** [comment]
retest this please

**Comment by VSadov** [comment]
@AlekseyTs  @gafter @jaredpar @agocke - please review

**Pull request merged**